File: backend/app/workers/jobs/llm.py
"""Feina programada: classificacio nocturna amb el model local, espai per espai."""

from __future__ import annotations

import logging

from sqlalchemy import select

from app.db import session_scope
from app.models import Ledger
from app.services.llm_classification import classify_merchants

logger = logging.getLogger(__name__)


def run_llm_classification(limit: int = 50) -> str:
    linies: list[str] = []
    with session_scope() as db:
        for ledger in db.scalars(select(Ledger).where(Ledger.is_active.is_(True))):
            stats = classify_merchants(db, ledger.id, limit=limit)
            linies.append(f"{ledger.name}: {stats}")
    resum = " | ".join(linies)
    logger.info("Classificacio amb model local: %s", resum)
    return resum

File: src/workers/jobs/llm.ts
/**
 * Feina programada: classificacio nocturna amb el model local, espai per espai.
 *
 * Es fa de nit perque en un NAS sense targeta grafica cada pregunta triga
 * segons. Traduccio de `backend/app/workers/jobs/llm.py`.
 */

import { eq } from "drizzle-orm";

import { db } from "../../db/client.ts";
import { ledgers } from "../../db/schema/index.ts";
import { classifyMerchants, summaryLlm } from "../../services/llm-classification.ts";

export async function localModelJob(limit = 50): Promise<string> {
  const linies: string[] = [];

  for (const workspace of await db.select().from(ledgers).where(eq(ledgers.isActive, true))) {
    const stats = await classifyMerchants(workspace.id, { limit });
    linies.push(`${workspace.name}: ${summaryLlm(stats)}`);
  }

  return linies.join("\n") || "no hi ha cap espai actiu";
}
